### ISSUE TITLE
Origins of Alpha

### ISSUE THREAD

**Issue by hyanwong** [jeromekelleher/sc2ts-paper#225]
This would be worth a line or two in the main text, and some digging for the supplementary. In the "maskreg-psv2-v1-mm_4-f500-clustloc-mrm_2-rw_10-mgs_10-2021-01-28" tree, it appears as if we can resolve the first of the Alpha mutations to matching against some samples sequenced in Cambridge (suspiciously geographically close to Kent, the origin of the alpha outbreak) in particular these samples, which are all B.1.1

```
{4598: 'ERR4413600', 6226: 'ERR4460507', 7812: 'ERR4460993', 1149: 'ERR4458709', 1150: 'ERR4458827', 8457: 'ERR4461558'}
```

<img width="1009" alt="Screenshot 2024-10-10 at 16 39 21" src="https://github.com/user-attachments/assets/5a256b3c-287a-4daf-84fd-453a01f0f1e7">

This is so early in the pandemic that we can easily re-run the matching up to that point. @jeromekelleher  points out that node 4725 here is likely to be caused by a reversion push, so it is a result of the tree-building part of the sc2ts algorithm. It has 3 exact matches:

```python
Node(id=4725, flags=4194304, time=318.250002625, population=-1, individual=-1, metadata={'sc2ts': {'date_added': '2020-03-31', 'num_exact_matches': 3, 'sites': [22909]}})
```

@szhan dug out a paper which also finds a Cambridge sample near the root of the alpha-defining lineages: https://www.ncbi.nlm.nih.gov/pmc/articles/PMC9752794/. We exclude their sample (CAMC-946506 / ERR4638413) because it has too much missing data: `sc2ts.inference Filter ERR4638413: missing=2393 > 500`, but it could be interesting to match it in afterwards.

Note that the taxonium tree below matches these 6 samples (circled) quite a long way from the root of alpha (alpha in cyan here). Shing suspects this is because of treating deletions as ancestral states.

![tmp](https://github.com/user-attachments/assets/39320c20-3952-49cd-af0a-e61e415437b1)

<details>
<summary>Code to replicate</summary >

```python
# will need to sub in any changed node IDs here
extra_nodes = [u for u in ts.at(21563).children(4725) if ts.node(u).is_sample()] + [u for u in ts.at(21563).samples(1151)]
svg = ti.draw_subtree(
    tracked_samples=[81660, 43835, 51309],  # Pick a few selected early samples from "B.1.1.7" / alpha
    size=(1100, 2000),
    canvas_size=(1100, 2000),
    time_scale="time",
    extra_tracked_samples=extra_nodes,
    style=".plotbox {transform: translateX(20px)}.leaf > .lab {text-anchor: start; transform: rotate(90deg) translate(6px)}",
)
svg
```
</details >

**Comment by jeromekelleher** [comment]
> 2024-10-10 15:02:22 DEBUG sc2ts.inference Final HMM pass hmm_cost=12.0 ERR4869224 2020-10-23 B.1.1 path=(0:29904, 4462) mutations(12)=[445T>C, 3264C>T, 5986C>T, 6286C>T, 6808T>C, 12247T>C, 15279C>T, 15775A>T, 23604C>A, 23709C>T, 25455G>T, 28977C>T]


It's from 2020-10-23, and I think it just was a bit too early to be within the window for inclusion. Would be interesting to see what would happen from matching it back later and adding it in all right.

**Comment by szhan** [comment]
Looking at the input sequence of ERR4869224, which has a decent number of Ns in these regions.
ORF1a:6866-7054 (which contains defining mutation T6954C:I2230T)
ORF1ab:21428-21458
S:22339-22523

I wonder how many imputed bases in at those sites (based on the best match) would be different than the ref. If all those Ns were supposed to be the same as ref., then we could have quite different placement in the trees.

**Comment by hyanwong** [comment]
The all-samples ARG is showing much the same patterns for Alpha, with a GAT->CTA triplet at the base of B.1.1.7 (cyan samples). We have removed the deletions here, but there are still lots of alpha-specific mutations, most of which are unique.

I'm suspicious about the C3267T recurrent mutation just below the alpha root (node 83297).


<img width="1724" alt="Screenshot 2024-10-21 at 15 34 08" src="https://github.com/user-attachments/assets/1ffacb15-c514-42a3-8289-b860b84236f6">

**Comment by jeromekelleher** [comment]
> I'm suspicious about the C3267T recurrent mutation just below the alpha root (node 83297).

Yes, this is odd, good spot. Let's keep an eye out for this on the next (final??) iteration. I'm suspicious that the HMM current settings will lead to slightly suboptimal HMM matches for really unlikely matches, and it may look something like this.

**Comment by jeromekelleher** [comment]
If you run the match and post the results we can figure out what inserting would do.

**Comment by hyanwong** [comment]
Certainly possible. Here's a version of the latest ARG, with some surrounding lineages expanded. Interesting that there is a decent recombination node associated with at the bottom left here (robust, but only one sample descendant, so could be a sequencing mixup).

Do we prefer the graph version or a conventional tree with right angles etc such as we get from draw_svg (above):

<img width="647" height="761" alt="Image" src="https://github.com/user-attachments/assets/34c33f9a-4158-45b9-9329-7de5dc63e687" />

**Comment by hyanwong** [comment]
Agree about the graph view. It would be nice to show which samples are based in the SE of England, but possibly too much faff (although I have some code to help with this).

The recombinant is annoying all right, but it does means we can say that we are capturing "recombination" events that haven't otherwise been spotted (whether or not these are because of sequencing mix-ups or true recombinants). It doesn't surprise me that it's stuck at the root, because the RHS of the recombinant is very short, so is probably identical, on that site, to the basal Alpha sequence (e.g. we can't tell whether it had that lone 3267 mutation).

Having this as a PDF would mean we can get the mutation positions on mouseover. otherwise I could print them as labels, but I would need to make it bigger to fit the labels in.

**Comment by hyanwong** [comment]
It would indeed be easy to split the PDF after it's produced into multiple pages, if that would help? I like having them in one document, but that's just personal preference. I think having all of them open in separate notebooks would be more likely to kill my browser than having a single notebook, though? Also if they are in one document, I can check at the end that we have created all the required subgraphs.

It takes a good number of minutes to import the ARG into the viz format, then more to set all the formatting etc. If we had to do that for each subgraph it would increase the time taken to get the output into many hours. There are a few other things that need doing, like reading in the consensus mutations and the data frames, but those should be quick.

So basically, it would be possible, but more work for me to change, and a considerably longer time to output all of them, which is why I would prefer not to.
